### ISSUE TITLE
AWS_PROFILE issue

### ISSUE THREAD

**Issue by fiftoine** [kubernetes-sigs/aws-iam-authenticator#201]
Hello,

TLDR; When using access and secret keys as env vars into kubeconfig I can access to a cluster but with aws configured with a profile using the same keys and using the AWS_PROFILE in kubeconfig, I cannot access the cluster.

Specifying AWS_PROFILE in aws-iam-authenticator does not seem to work.

I configured aws cli like this
```
[my-other-profile]
aws_access_key_id = [redacted]
aws_secret_access_key = [redacted]
```

From that profile I can describe the cluster
```
$ aws eks describe-cluster --name my-cluster --profile my-other-profile
{
    "cluster": {
...
    }
}
```

If I configure my kubeconfig file like this:
```
- name: my-cluster-user
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      env:
      - name: "AWS_PROFILE"
        value: "my-other-profile"
      args:
      - token
      - -i
      - my-cluster
      command: aws-iam-authenticator
```

When I try to access pods : 
```
$ kubectl auth can-i get pods
error: You must be logged in to the server (Unauthorized)
```

But if I put access and secret key as environment variables in the kubeconfig like so : 
```
- name: my-cluster-user
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      env:
      - name: "AWS_ACCESS_KEY_ID"
        value: "[redacted]"
      - name: "AWS_SECRET_ACCESS_KEY"
        value: "[redacted]"
      args:
      - token
      - -i
      - my-cluster
      command: aws-iam-authenticator
```

I got the result : 
```
$ kubectl auth can-i get pods
yes
```

Is this a bug or am I doing something wrong?

Thanks

**Comment by mjstrasser** [comment]
I have a similar setup with AWS_PROFILE in the shell environment. My kubeconfig includes:
```yaml
users:
- name: aws
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      command: aws-iam-authenticator
      args:
      - "token"
      - "-i"
      - "my-cluster"
```
Then in my shell:
```bash
$ export AWS_PROFILE=my_profile
$ kubectl auth can-i get pods
yes
```
Hope this helps.

**Comment by nckturner** [comment]
@fiftoine did you ever get this to work?  You should be able to specify the profile like you are doing with: 
```
      env:
      - name: "AWS_PROFILE"
        value: "my-other-profile"
```
